modules/localstack/localstack.go

```diff
@@ -24,7 +24,11 @@ func isLegacyMode(image string) bool {
 	parts := strings.Split(image, ":")
 	version := parts[len(parts)-1]
 
-	if version == "latest" {
+	if pos := strings.LastIndexByte(version, '-'); pos >= 0 {
+		version = version[0:pos]
+	}
+
+	if version == "latest" || version == "s3" || version == "s3-latest" || version == "stable" {
```

Comment:
suggestion: we could add an slice of valid non-legacy versions at the top of the file. Here we would just look up the used version to verify if it's legacy. Wdyt?

Reply:
Are you suggesting every legacy version is simply hard coded in a slice? It looks like there are 31 versions prior to `v0.11`, plus the following unconventional tags that I have no idea about:

- `2021-11-15`
- `2022-01-27`
- `2022-01-28`
- `2022-07-25`
- `fix-perf`
- `jvm-2022-11-03`
- `m1-jre-fix`
- `multi`

Reply:
Here's a much simpler implementation that actually works (except for the junk "foo" test case).

```go
func isLegacyMode(image string) bool {
	return strings.Compare(image, "localstack/localstack:0.11") < 0
}
```

Reply:
> Are you suggesting every legacy version is simply hard coded in a slice? It looks like there are 31 versions prior to v0.11, plus the following unconventional tags that I have no idea aboutAre you suggesting every legacy version

No, just the ones in the if branch:
```
if version == "latest" || version == "s3" || version == "s3-latest" || version == "stable"
```

Reply:
Sure, done. I've also fixed the `isVersion2` function along with unit tests.